addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3315,6 +3315,7 @@ const ImageHandlerOption = SnippetOptionWidget.extend({
         $weight.find('b').text(`${(weight / 1024).toFixed(1)} kb`);
         $weight.removeClass('d-none');
         img.classList.add('o_modified_image_to_save');
+        this._forceOptions(img);
```

Comment:
What I had in mind here was to move all the contents of that `_applyOptions` in the `_forceOptions` method, except the "if optimizable" check. Then extend the method for the background image.

But what you have done is actually also good and in fact allows to find a better name than `_forceOptions`. Name it `_applyImage`: in the normal case no image is to be applied since applyOption works with the image in the DOM = empty method like you made, and in the background image option, the image whose options have just been applied has to be applied as background image.